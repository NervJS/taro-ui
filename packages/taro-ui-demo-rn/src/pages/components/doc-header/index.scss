.doc-header {
  display: flex;
  flex-direction: row;
  align-items: center;
  padding: 60px;
  height: 180px;
  position: relative;

  &__title {
    height: 90px;
    color: #333;
    font-size: 40px;
    font-weight: bold;

    /*  #ifndef  rn  */
    &::after {
      content: '';
      position: absolute;
      left: 0;
      bottom: 0;
      display: inline-block;
      width: 120px;
      height: 2PX;
      border-radius: 1PX;
      background-color: #6190E8;
    }
    /*  #endif  */
  }

  &__line {
    position: absolute;
    left: 60px;
    bottom: 50px;
    display: flex;
    width: 120px;
    height: 2PX;
    border-radius: 1PX;
    background-color: #6190E8;
  }
}
