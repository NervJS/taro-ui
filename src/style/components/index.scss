/**
 * Components
 */

@import './icon.scss';
@import './flex.scss';
@import './accordion.scss';
@import './activity-indicator.scss';
@import './action-sheet.scss';
@import './article.scss';
@import './avatar.scss';
@import './badge.scss';
@import './button.scss';
@import './calendar.scss';
@import './card.scss';
@import './checkbox.scss';
@import './countdown.scss';
@import './curtain.scss';
@import './divider.scss';
@import './drawer.scss';
@import './fab.scss';
@import './float-layout.scss';
@import './form.scss';
@import './grid.scss';
@import './image-picker.scss';
@import './indexes.scss';
@import './input-number.scss';
@import './input.scss';
@import './list.scss';
@import './load-more.scss';
@import './loading.scss';
@import './message.scss';
@import './modal.scss';
@import './nav-bar.scss';
@import './noticebar.scss';
@import './pagination.scss';
@import './progress.scss';
@import './radio.scss';
@import './range.scss';
@import './rate.scss';
@import './search-bar.scss';
@import './segmented-control.scss';
@import './slider.scss';
@import './steps.scss';
@import './swipe-action.scss';
@import './switch.scss';
@import './tab-bar.scss';
@import './tabs.scss';
@import './tag.scss';
@import './textarea.scss';
@import './timeline.scss';
@import './toast.scss';
