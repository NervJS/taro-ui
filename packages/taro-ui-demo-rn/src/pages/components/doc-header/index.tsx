import PropTypes, { InferProps } from 'prop-types'
import React from 'react'
import { View, Text } from '@tarojs/components'
import './index.scss'

export interface DocsHeaderProps {
  title?: string
}

export default class DocsHeader extends React.Component<DocsHeaderProps> {
  public static defaultProps: DocsHeaderProps
  public static propTypes: InferProps<DocsHeaderProps>

  public render(): JSX.Element {
    const { title } = this.props

    return (
      <View className='doc-header'>
        <Text className='doc-header__title'>{title}</Text>
        {process.env.TARO_ENV === 'rn' && <View className='doc-header__line' />}
      </View>
    )
  }
}

DocsHeader.defaultProps = {
  title: '标题'
}

DocsHeader.propTypes = {
  title: PropTypes.string
}
