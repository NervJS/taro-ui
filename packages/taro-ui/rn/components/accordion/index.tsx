import classNames from 'classnames'
import PropTypes, { InferProps } from 'prop-types'
import React from 'react'
import { Text, View } from '@tarojs/components'
import { CommonEvent } from '@tarojs/components/types/common'
import { AtAccordionProps, AtAccordionState } from '../../../types/accordion'
import { delayQuerySelector } from '../../common/utils'

export default class AtAccordion extends React.Component<
  AtAccordionProps,
  AtAccordionState
> {
  private isCompleted: boolean
  private startOpen: boolean

  public static defaultProps: AtAccordionProps
  public static propTypes: InferProps<AtAccordionProps>

  public constructor(props: AtAccordionProps) {
    super(props)
    this.isCompleted = true
    this.startOpen = false
    this.state = {
      wrapperHeight: 0
    }
  }

  private handleClick = (event: CommonEvent): void => {
    const { open } = this.props
    if (!this.isCompleted) return

    this.props.onClick && this.props.onClick(!open, event)
  }

  private toggleWithAnimation(): void {
    const { open, isAnimation } = this.props
    if (!this.isCompleted || !isAnimation) return

    this.isCompleted = false
    delayQuerySelector('.at-accordion__body', 0).then(rect => {
      const height = parseInt(rect[0].height.toString())
      const startHeight = open ? height : 0
      const endHeight = open ? 0 : height
      this.startOpen = false
      this.setState(
        {
          wrapperHeight: startHeight
        },
        () => {
          setTimeout(() => {
            this.setState(
              {
                wrapperHeight: endHeight
              },
              () => {
                setTimeout(() => {
                  this.isCompleted = true
                  this.setState({})
                }, 700)
              }
            )
          }, 100)
        }
      )
    })
  }

  public UNSAFE_componentWillReceiveProps(nextProps: AtAccordionProps): void {
    if (nextProps.open !== this.props.open) {
      this.startOpen = !!nextProps.open && !!nextProps.isAnimation
      this.toggleWithAnimation()
    }
  }

  public render(): JSX.Element {
    const {
      customStyle,
      className,
      title,
      icon,
      hasBorder,
      open,
      note
    } = this.props
    const { wrapperHeight } = this.state

    const rootCls = classNames('at-accordion', className)
    const prefixClass = (icon && icon.prefixClass) || 'at-icon'
    const iconCls = classNames({
      [prefixClass]: true,
      [`${prefixClass}-${icon && icon.value}`]: icon && icon.value,
      'at-accordion__icon': true
    })
    const headerCls = classNames('at-accordion__header', {
      'at-accordion__header--noborder': !hasBorder
    })
    const arrowCls = classNames('at-accordion__arrow', {
      'at-accordion__arrow--folded': !!open
    })
    const contentCls = classNames('at-accordion__content', {
      'at-accordion__content--inactive':
        (!open && this.isCompleted) || this.startOpen
    })
    const iconStyle = {
      color: (icon && icon.color) || '',
      fontSize: (icon && `${icon.size}px`) || ''
    }
    const contentStyle = { height: `${wrapperHeight}px` }

    if (this.isCompleted) {
      contentStyle.height = ''
    }

    return (
      <View className={rootCls} style={customStyle}>
        <View className={headerCls} onClick={this.handleClick}>
          {icon && icon.value && (
            <Text className={iconCls} style={iconStyle}></Text>
          )}
          <View className='at-accordion__info'>
            <View className='at-accordion__info__title'>{title}</View>
            <View className='at-accordion__info__note'>{note}</View>
          </View>
          <View className={arrowCls}>
            <Text className='at-icon at-icon-chevron-down'></Text>
          </View>
        </View>
        <View style={contentStyle} className={contentCls}>
          <View className='at-accordion__body'>{this.props.children}</View>
        </View>
      </View>
    )
  }
}

AtAccordion.defaultProps = {
  open: false,
  customStyle: '',
  className: '',
  title: '',
  note: '',
  icon: { value: '' },
  hasBorder: true,
  isAnimation: true
}

AtAccordion.propTypes = {
  customStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
  className: PropTypes.oneOfType([PropTypes.array, PropTypes.string]),
  open: PropTypes.bool,
  isAnimation: PropTypes.bool,
  title: PropTypes.string,
  note: PropTypes.string,
  icon: PropTypes.object,
  hasBorder: PropTypes.bool,
  onClick: PropTypes.func
}
