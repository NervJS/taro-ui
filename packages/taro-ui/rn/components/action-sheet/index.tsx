import classNames from 'classnames'
import PropTypes, { InferProps } from 'prop-types'
import React from 'react'
import { View } from '@tarojs/components'
import Modal from 'react-native-modal'
import { CommonEvent } from '@tarojs/components/types/common'
import {
  AtActionSheetProps,
  AtActionSheetState,
} from '../../../types/action-sheet'
import AtActionSheetBody from './body/index'
import AtActionSheetFooter from './footer/index'
import AtActionSheetHeader from './header/index'

export default class AtActionSheet extends React.Component<
  AtActionSheetProps,
  AtActionSheetState
> {
  public static defaultProps: AtActionSheetProps
  public static propTypes: InferProps<AtActionSheetProps>

  public constructor(props: AtActionSheetProps) {
    super(props)
    const { isOpened } = props

    this.state = {
      _isOpened: isOpened,
    }
  }

  public UNSAFE_componentWillReceiveProps(nextProps: AtActionSheetProps): void {
    const { isOpened } = nextProps
    if (isOpened !== this.state._isOpened) {
      this.setState({
        _isOpened: isOpened,
      })

      !isOpened && this.handleClose()
    }
  }

  private handleClose = (): void => {
    if (typeof this.props.onClose === 'function') {
      this.props.onClose()
    }
  }

  private handleCancel = (): void => {
    if (typeof this.props.onCancel === 'function') {
      return this.props.onCancel()
    }
    this.close()
  }

  private close = (): void => {
    this.setState(
      {
        _isOpened: false,
      },
      this.handleClose,
    )
  }

  private handleTouchMove = (e: CommonEvent): void => {
    e.stopPropagation()
    e.preventDefault()
  }

  public render(): JSX.Element {
    const { title, cancelText, className } = this.props
    const { _isOpened } = this.state

    // const rootClass = classNames(
    //   'at-action-sheet',
    //   {
    //     'at-action-sheet--active': _isOpened
    //   },
    //   className
    // )
    const containerClass = classNames('at-action-sheet__container', className)

    return (
      <Modal
        animationIn='fadeInUp'
        animationOut='fadeOutDown'
        isVisible={_isOpened}
        hasBackdrop
        onModalHide={this.handleClose}
        onBackButtonPress={this.handleClose}
        onBackdropPress={this.handleClose}
        style={{
          margin: 0,
        }}
      >
        <View className={containerClass}>
          {title && <AtActionSheetHeader>{title}</AtActionSheetHeader>}
          <AtActionSheetBody>{this.props.children}</AtActionSheetBody>
          {cancelText && (
            <AtActionSheetFooter onClick={this.handleCancel}>
              {cancelText}
            </AtActionSheetFooter>
          )}
        </View>
      </Modal>
    )
  }
}

AtActionSheet.defaultProps = {
  title: '',
  cancelText: '',
  isOpened: false,
}

AtActionSheet.propTypes = {
  title: PropTypes.string,
  onClose: PropTypes.func,
  onCancel: PropTypes.func,
  isOpened: PropTypes.bool.isRequired,
  cancelText: PropTypes.string,
}
