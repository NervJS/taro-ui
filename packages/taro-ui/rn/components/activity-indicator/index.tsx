import classNames from 'classnames'
import PropTypes, { InferProps } from 'prop-types'
import React from 'react'
import { Text, View } from '@tarojs/components'
import { AtActivityIndicatorProps } from '../../../types/activity-indicator'
import AtLoading from '../loading/index'

export default class AtActivityIndicator extends React.Component<
  AtActivityIndicatorProps
> {
  public static defaultProps: AtActivityIndicatorProps
  public static propTypes: InferProps<AtActivityIndicatorProps>

  public render(): JSX.Element {
    const { color, size, mode, content, isOpened } = this.props

    const rootClass = classNames(
      'at-activity-indicator',
      {
        'at-activity-indicator--center': mode === 'center',
        'at-activity-indicator--isopened': isOpened
      },
      this.props.className
    )

    return (
      <View className={rootClass}>
        <View className='at-activity-indicator__body'>
          <AtLoading size={size} color={color} />
        </View>
        {content && (
          <Text className='at-activity-indicator__content'>{content}</Text>
        )}
      </View>
    )
  }
}

AtActivityIndicator.defaultProps = {
  size: 0,
  mode: 'normal',
  color: '',
  content: '',
  className: '',
  isOpened: true
}

AtActivityIndicator.propTypes = {
  size: PropTypes.number,
  mode: PropTypes.string,
  color: PropTypes.string,
  content: PropTypes.string,
  className: PropTypes.oneOfType([PropTypes.array, PropTypes.string]),
  isOpened: PropTypes.bool
}
