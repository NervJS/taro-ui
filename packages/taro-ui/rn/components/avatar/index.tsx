import classNames from 'classnames'
import PropTypes, { InferProps } from 'prop-types'
import React from 'react'
import { Image, View } from '@tarojs/components'
import Taro from '@tarojs/taro'
import { AtAvatarProps, AtAvatarState } from '../../../types/avatar'

let OpenData: any = null
if (process.env.TARO_ENV === 'rn') {
  OpenData = View
} else {
  OpenData = require('@tarojs/components')
}

const SIZE_CLASS = {
  large: 'large',
  normal: 'normal',
  small: 'small',
}

export default class AtAvatar extends React.Component<
  AtAvatarProps,
  AtAvatarState
> {
  public static defaultProps: AtAvatarProps
  public static propTypes: InferProps<AtAvatarProps>

  public constructor(props: AtAvatarProps) {
    super(props)
    this.state = {
      isWEAPP: Taro.getEnv() === Taro.ENV_TYPE.WEAPP,
      _width: 0,
    }
  }

  private onLayout = (event: any): void => {
    const { width } = event.nativeEvent.layout
    this.setState({ _width: width })
  }

  public render(): JSX.Element {
    const { _width } = this.state
    const { size, circle, image, text, openData, customStyle } = this.props
    const rootClassName = ['at-avatar']
    const iconSize = SIZE_CLASS[size || 'normal']
    const classObject = {
      [`at-avatar--${iconSize}`]: iconSize,
      'at-avatar--circle': circle,
    }

    let letter = ''
    if (text) letter = text[0]

    let elem: React.ReactNode
    if (openData && openData.type === 'userAvatarUrl' && this.state.isWEAPP) {
      elem = <OpenData type={openData.type}></OpenData>
    } else if (image) {
      elem = <Image className='at-avatar__img' src={image} />
    } else {
      // elem = <Text className='at-avatar__text'>{letter}</Text>
      // TODO: RN
      elem = letter
    }

    let style = customStyle
    if (typeof customStyle === 'object' && _width && circle) {
      style = { ...customStyle, borderRadius: _width / 2 }
    }

    return (
      <View
        className={classNames(rootClassName, classObject, this.props.className)}
        style={style}
        onLayout={this.onLayout}
      >
        {elem}
      </View>
    )
  }
}

AtAvatar.defaultProps = {
  size: 'normal',
  circle: false,
  text: '',
  image: '',
  customStyle: {},
  className: '',
}

AtAvatar.propTypes = {
  size: PropTypes.oneOf(['large', 'normal', 'small']),
  circle: PropTypes.bool,
  text: PropTypes.string,
  image: PropTypes.string,
  openData: PropTypes.object,
  customStyle: PropTypes.oneOfType([PropTypes.object]),
  className: PropTypes.oneOfType([PropTypes.array, PropTypes.string]),
}
