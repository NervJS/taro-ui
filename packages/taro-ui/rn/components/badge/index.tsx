import classNames from 'classnames'
import PropTypes, { InferProps } from 'prop-types'
import React from 'react'
import { View } from '@tarojs/components'
import { AtBadgeProps } from '../../../types/badge'

export default class AtBadge extends React.Component<AtBadgeProps> {
  public static defaultProps: AtBadgeProps
  public static propTypes: InferProps<AtBadgeProps>

  public constructor(props: AtBadgeProps) {
    super(props)
    this.state = {}
  }

  private formatValue(
    value: string | number | undefined,
    maxValue: number
  ): string | number {
    if (value === '' || value === null || typeof value === 'undefined')
      return ''
    const numValue = +value
    if (Number.isNaN(numValue)) {
      return value
    }
    return numValue > maxValue ? `${maxValue}+` : numValue
  }

  public render(): JSX.Element {
    const { dot, value, maxValue = 99, customStyle } = this.props
    const rootClassName = ['at-badge']

    const val = this.formatValue(value, maxValue)

    return (
      <View
        className={classNames(rootClassName, this.props.className)}
        style={customStyle}
      >
        {this.props.children}
        {dot ? (
          <View className='at-badge__dot'></View>
        ) : (
          val !== '' && <View className='at-badge__num'>{val}</View>
        )}
      </View>
    )
  }
}

AtBadge.defaultProps = {
  dot: false,
  value: '',
  maxValue: 99,
  customStyle: {},
  className: ''
}

AtBadge.propTypes = {
  dot: PropTypes.bool,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  maxValue: PropTypes.number,
  customStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
  className: PropTypes.oneOfType([PropTypes.array, PropTypes.string])
}
