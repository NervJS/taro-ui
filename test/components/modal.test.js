import Nerv, { findDOMNode } from 'nervjs'
import { renderToString } from 'nerv-server'
import { Simulate, renderIntoDocument } from 'nerv-test-utils'
import { Button } from '@tarojs/components'
import AtModal from '../../.temp/components/modal/index'
import AtModalAction from '../../.temp/components/modal/action/index'
import AtModalHeader from '../../.temp/components/modal/header/index'
import AtModalContent from '../../.temp/components/modal/content/index'

describe('Modal Snap', () => {
  it('render initial Modal', () => {
    const component = renderToString(
      <AtModal>
        <AtModalHeader>标题</AtModalHeader>
        <AtModalContent>
          这里是正文内容，欢迎加入京东凹凸实验室
          这里是正文内容，欢迎加入京东凹凸实验室
          这里是正文内容，欢迎加入京东凹凸实验室
        </AtModalContent>
        <AtModalAction>
          <Button>取消</Button>
          <Button>确定</Button>
        </AtModalAction>
      </AtModal>
    )
    expect(component).toMatchSnapshot()
  })

  it('render opened Modal', () => {
    const component = renderToString(
      <AtModal isOpened>
        <AtModalHeader>标题</AtModalHeader>
        <AtModalContent>
          这里是正文内容，欢迎加入京东凹凸实验室
          这里是正文内容，欢迎加入京东凹凸实验室
          这里是正文内容，欢迎加入京东凹凸实验室
        </AtModalContent>
        <AtModalAction>
          <Button>取消</Button>
          <Button>确定</Button>
        </AtModalAction>
      </AtModal>
    )
    expect(component).toMatchSnapshot()
  })

  it('render opened Modal -- not header', () => {
    const component = renderToString(
      <AtModal isOpened>
        <AtModalContent>
          这里是正文内容，欢迎加入京东凹凸实验室
          这里是正文内容，欢迎加入京东凹凸实验室
          这里是正文内容，欢迎加入京东凹凸实验室
        </AtModalContent>
        <AtModalAction>
          <Button>取消</Button>
          <Button>确定</Button>
        </AtModalAction>
      </AtModal>
    )
    expect(component).toMatchSnapshot()
  })

  it('render opened Modal -- single button', () => {
    const component = renderToString(
      <AtModal isOpened>
        <AtModalContent>
          这里是正文内容，欢迎加入京东凹凸实验室
          这里是正文内容，欢迎加入京东凹凸实验室
          这里是正文内容，欢迎加入京东凹凸实验室
        </AtModalContent>
        <AtModalAction>
          <Button>确定</Button>
        </AtModalAction>
      </AtModal>
    )
    expect(component).toMatchSnapshot()
  })

  it('render opened  Modal -- simple', () => {
    const component = renderToString(
      <AtModal
        isOpened
        title='标题'
        cancelText='取消'
        confirmText='确认'
        content='欢迎加入京东凹凸实验室\n\r欢迎加入京东凹凸实验室'
      />
    )
    expect(component).toMatchSnapshot()
  })
})

describe('Modal Behavior ', () => {
  it('Modal onClose & onCancel & onClick', () => {
    const onCancel = jest.fn()
    const onConfirm = jest.fn()
    const onClose = jest.fn()

    const component = renderIntoDocument(
      <AtModal
        isOpened
        title='标题'
        cancelText='取消'
        confirmText='确认'
        onClose={onClose}
        onCancel={onCancel}
        onConfirm={onConfirm}
        content='欢迎加入京东凹凸实验室\n\r欢迎加入京东凹凸实验室'
      />
    )
    const componentDom = findDOMNode(component, 'at-modal')

    const cancelDom = componentDom.querySelector(
      '.at-modal__footer .at-modal__action button:first-child'
    )
    const confirmDom = componentDom.querySelector(
      '.at-modal__footer .at-modal__action button:last-child'
    )
    const overlayDom = componentDom.querySelector('.at-modal__overlay')

    Simulate.click(cancelDom)
    expect(onCancel).toBeCalled()

    Simulate.click(confirmDom)
    expect(onConfirm).toBeCalled()

    expect(component.state._isOpened).toBeTruthy()
    Simulate.click(overlayDom)
    process.nextTick(() => {
      expect(onClose).toBeCalled()
      expect(component.state._isOpened).toBeFalsy()
    })
  })

  it('Modal onClose will not be called', () => {
    const onClose = jest.fn()

    const component = renderIntoDocument(
      <AtModal
        isOpened
        title='标题'
        cancelText='取消'
        closeOnClickOverlay={false}
        confirmText='确认'
        onClose={onClose}
        content='欢迎加入京东凹凸实验室\n\r欢迎加入京东凹凸实验室'
      />
    )
    const componentDom = findDOMNode(component, 'at-modal')

    const overlayDom = componentDom.querySelector('.at-modal__overlay')

    expect(component.state._isOpened).toBeTruthy()

    Simulate.click(overlayDom)

    process.nextTick(() => {
      expect(onClose).not.toBeCalled()
      expect(component.state._isOpened).toBeTruthy()
    })
  })
})
