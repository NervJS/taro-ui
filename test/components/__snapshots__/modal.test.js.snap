// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Snap render initial Modal 1`] = `"<div class=\\"at-modal\\"><div class=\\"at-modal__overlay\\"></div><div class=\\"at-modal__container\\"><div class=\\"at-modal__header\\">标题</div><div scrollY class=\\"taro-scroll taro-scroll-view__scroll-y at-modal__content\\">这里是正文内容，欢迎加入京东凹凸实验室 这里是正文内容，欢迎加入京东凹凸实验室 这里是正文内容，欢迎加入京东凹凸实验室</div><div class=\\"at-modal__footer\\"><div class=\\"at-modal__action\\"><button class=\\"weui-btn weui-btn_default\\">undefined取消</button><button class=\\"weui-btn weui-btn_default\\">undefined确定</button></div></div></div></div>"`;

exports[`Modal Snap render opened  Modal -- simple 1`] = `"<div class=\\"at-modal at-modal--active\\"><div class=\\"at-modal__overlay\\"></div><div class=\\"at-modal__container\\"><div class=\\"at-modal__header\\"><span class=\\"taro-text\\">标题</span></div><div scrollY class=\\"taro-scroll taro-scroll-view__scroll-y at-modal__content\\"><div class=\\"content-simple\\"><span class=\\"taro-text\\">欢迎加入京东凹凸实验室\\\\n\\\\r欢迎加入京东凹凸实验室</span></div></div><div class=\\"at-modal__footer at-modal__footer--simple\\"><div class=\\"at-modal__action\\"><button class=\\"weui-btn weui-btn_default\\">undefined取消</button><button class=\\"weui-btn weui-btn_default\\">undefined确认</button></div></div></div></div>"`;

exports[`Modal Snap render opened Modal -- not header 1`] = `"<div class=\\"at-modal at-modal--active\\"><div class=\\"at-modal__overlay\\"></div><div class=\\"at-modal__container\\"><div scrollY class=\\"taro-scroll taro-scroll-view__scroll-y at-modal__content\\">这里是正文内容，欢迎加入京东凹凸实验室 这里是正文内容，欢迎加入京东凹凸实验室 这里是正文内容，欢迎加入京东凹凸实验室</div><div class=\\"at-modal__footer\\"><div class=\\"at-modal__action\\"><button class=\\"weui-btn weui-btn_default\\">undefined取消</button><button class=\\"weui-btn weui-btn_default\\">undefined确定</button></div></div></div></div>"`;

exports[`Modal Snap render opened Modal -- single button 1`] = `"<div class=\\"at-modal at-modal--active\\"><div class=\\"at-modal__overlay\\"></div><div class=\\"at-modal__container\\"><div scrollY class=\\"taro-scroll taro-scroll-view__scroll-y at-modal__content\\">这里是正文内容，欢迎加入京东凹凸实验室 这里是正文内容，欢迎加入京东凹凸实验室 这里是正文内容，欢迎加入京东凹凸实验室</div><div class=\\"at-modal__footer\\"><div class=\\"at-modal__action\\"><button class=\\"weui-btn weui-btn_default\\">undefined确定</button></div></div></div></div>"`;

exports[`Modal Snap render opened Modal 1`] = `"<div class=\\"at-modal at-modal--active\\"><div class=\\"at-modal__overlay\\"></div><div class=\\"at-modal__container\\"><div class=\\"at-modal__header\\">标题</div><div scrollY class=\\"taro-scroll taro-scroll-view__scroll-y at-modal__content\\">这里是正文内容，欢迎加入京东凹凸实验室 这里是正文内容，欢迎加入京东凹凸实验室 这里是正文内容，欢迎加入京东凹凸实验室</div><div class=\\"at-modal__footer\\"><div class=\\"at-modal__action\\"><button class=\\"weui-btn weui-btn_default\\">undefined取消</button><button class=\\"weui-btn weui-btn_default\\">undefined确定</button></div></div></div></div>"`;
